force-app/main/adapter/in/sobjects/contact/Contacts.cls

```diff
@@ -539,7 +539,7 @@ public inherited sharing class Contacts extends fflib_SObjects {
         for (Contact contact : listCon) {
             NPSP_Address npspAddress = new NPSP_Address(addressesByContact.get(contact));
             NPSP_Address npspMatchingAddress = new NPSP_Address(addressMatchesByAddress.get(npspAddress.getRecord()));
-            NPSP_Contact npspContact = new NPSP_Contact(contact);
+            NPSP_Contact npspContact = new NPSP_Contact(contact, oldVersionOf(contact));
```

Comment:
This doesn't seem to hurt anything, but are we using the old Contact anywhere?

Reply:
Yeah we actually need that old contact version, particularly when we are checking if the undeliverable status on the contact record has changed in the `isUndeliverableStatusChanged()` method of `NPSP_Contact`. The absence of this parameter in the constructor is what caused the regression I called out in the design doc.